pkg/identity/identity.go

```diff
@@ -58,15 +58,16 @@ type Identity struct {
 // This structure is written as JSON to the key-value store. Do NOT modify this
 // structure in ways which are not JSON forward compatible.
 type IPIdentityPair struct {
-	IP           net.IP          `json:"IP"`
-	Mask         net.IPMask      `json:"Mask"`
-	HostIP       net.IP          `json:"HostIP"`
-	ID           NumericIdentity `json:"ID"`
-	Key          uint8           `json:"Key"`
-	Metadata     string          `json:"Metadata"`
-	K8sNamespace string          `json:"K8sNamespace,omitempty"`
-	K8sPodName   string          `json:"K8sPodName,omitempty"`
-	NamedPorts   []NamedPort     `json:"NamedPorts,omitempty"`
+	IP                net.IP          `json:"IP"`
+	Mask              net.IPMask      `json:"Mask"`
+	HostIP            net.IP          `json:"HostIP"`
+	ID                NumericIdentity `json:"ID"`
+	Key               uint8           `json:"Key"`
+	Metadata          string          `json:"Metadata"`
+	K8sNamespace      string          `json:"K8sNamespace,omitempty"`
+	K8sPodName        string          `json:"K8sPodName,omitempty"`
+	K8sServiceAccount string          `json:"K8sServiceAccount,omitempty"`
```

Comment:
The new SA field needs to be set in the clustermesh-apiserver codepaths as well for consistency, even though not currently used:

* https://github.com/cilium/cilium/blob/1af8ac859885d42fdd5a6024b21c47abb4ccd64f/clustermesh-apiserver/clustermesh/converters.go#L188-L193
* https://github.com/cilium/cilium/blob/1af8ac859885d42fdd5a6024b21c47abb4ccd64f/clustermesh-apiserver/clustermesh/converters.go#L235-L242

Reply:
Great! I'll update, thanks for this info. 

I'm also noticing some text files which seem to provide test data, for example: https://github.com/cilium/cilium/blob/405543d15824ad9fbd22fe7adf5c1c5a2353a3e7/clustermesh-apiserver/clustermesh/testdata/ciliumendpoints.txtar#L43-L100

Should I update these test files with the new service-account file as well, for CEP and CES?

Reply:
Done, ran tests and they seem good.